language/move-prover/tests/sources/functional/verify_vector.move

```diff
@@ -291,6 +291,23 @@ module 0x42::VerifyVector {
         ensures old(v[i]) == result;
     }
 
+    // Add an element to the vector at index.
+    fun verify_model_insert<Element>(v: &mut vector<Element>, e: Element, i: u64) {
+        vector::insert(v, e, i); // inlining the built-in Boogie procedure
+    }
+    spec verify_model_insert {
+        aborts_if i > len(v);
+        ensures len(v) == len(old(v)) + 1;
+        ensures v[i] == e;
+        ensures old(v[0..i]) == v[0..i];
+        // the following ensures is so complicated because we cannot use len(old(v)) in a nested
```

Comment:
The following ensures looks complicated. By the way, what is it trying to achieve?

Reply:
Indeed. It has been simplified in the newer version thanks to @meng-xu-cs's suggestion!